modules/imgproc/src/drawing.cpp

```diff
@@ -1875,13 +1875,12 @@ void rectangle( InputOutputArray _img, Point pt1, Point pt2,
 }
 
 
-void rectangle( Mat& img, Rect rec,
+void rectangle( InputOutputArray img, Rect rec,
                 const Scalar& color, int thickness,
                 int lineType, int shift )
 {
     CV_INSTRUMENT_REGION()
 
-    CV_Assert( 0 <= shift && shift <= XY_SHIFT );
```

Comment:
Why is assertion removed?

Reply:
maybe i am wrong but i think no need because of the same assertion exist in called function

https://github.com/sturkmen72/opencv/blob/master/modules/imgproc/src/drawing.cpp#L1855